app/jobs/event/participation_cleanup_answers_job.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito.
+
+class Event::ParticipationCleanupAnswersJob < RecurringJob
+  run_every 1.day
+
+  private
+
+  def perform_internal
+    obsolete_answers.delete_all if cutoff
+  end
+
+  def obsolete_answers
+    Event::Answer.joins(:participation).where(not_participating_since_cutoff)
+  end
+
+  def not_participating_since_cutoff
+    Event.joins(:dates)
```

Comment:
Das bräuchte ein where not exists anstatt join where. In der aktuellen Version wird eine Event-Antwort gefunden sobald x Monate seit dem ersten (statt letzten) Event Date vergangen sind

Reply:
Ich denke das stimmt so schon, zumindest gemäss den tests .. wir joinen aber machen ein `not.exist` in der subquery, i.e 

```
SELECT `event_answers`.* FROM `event_answers` 
INNER JOIN `event_participations` ON `event_participations`.`id` = `event_answers`.`participation_id` 
WHERE NOT (EXISTS (SELECT `events`.`id` FROM `events` INNER JOIN `event_dates` ON `event_dates`.`event_id` = `events`.`id` 
  WHERE (event_dates.start_at > '2012-02-28 23:00:00' OR event_dates.finish_at > '2012-02-28 23:00:00') 
  AND (event_participations.event_id = events.id)
 )
)
```

hab auch noch einen test ergänzt, der das mit mehreen Tagen testen